File: minimum-increment-to-make-array-unique.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        nums.sort()
        increment = 0
        for i in range(1, len(nums)):
            if nums[i] <= nums[i - 1]:
                diff = nums[i - 1] - nums[i] + 1
                nums[i] += diff
                increment += diff
        return increment

File: c++/minimum-increment-to-make-array-unique.cc
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int counter = 0;
        int curLargest = -1;

        for (int num : nums) {
            if (num > curLargest) {
                curLargest = num;
            } else {
                counter += curLargest - num + 1;
                curLargest++;
            }
        }

        return counter;
    }
};